PostHog/PostHogRemoteConfig.swift

```diff
@@ -1,26 +1,40 @@
 //
-//  PostHogFeatureFlags.swift
+//  PostHogRemoteConfig.swift
 //  PostHog
 //
 //  Created by Manoel Aranda Neto on 10.10.23.
 //
 
 import Foundation
 
-class PostHogFeatureFlags {
+class PostHogRemoteConfig {
+    private let hasFeatureFlagsKey = "hasFeatureFlags"
+
     private let config: PostHogConfig
     private let storage: PostHogStorage
     private let api: PostHogApi
 
-    private let loadingLock = NSLock()
+    private let loadingFeatureFlagsLock = NSLock()
     private let featureFlagsLock = NSLock()
     private var loadingFeatureFlags = false
     private var sessionReplayFlagActive = false
-
     private var featureFlags: [String: Any]?
+
+    private var remoteConfigLock = NSLock()
+    private let loadingRemoteConfigLock = NSLock()
+    private var loadingRemoteConfig = false
+    private var remoteConfig: [String: Any]?
+    private var hasFeatureFlags: Bool?
```

Comment:
do we need a global state for that? i think we can just check the return from the API and reload flags if there's any

Reply:
If someone calls `reloadFeatureFlags` manually, the intention is to fetch the API because something might have changed. In this case, we should not check what's cached in memory.